src/components/work/TripCard/TripCard.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import { Card, CardImg, CardBody, CardTitle, CardSubtitle, Row, Col, CardText, CardFooter } from 'reactstrap';
+import { Booking } from 'networking/bookings';
+import { getUserBookingDisplayStatus } from 'utils/bookingsDisplayStatus';
+import { formatAddress, formatGeolocationAddress } from 'utils/formatter';
+import { formatSingleDate } from 'utils/formatDate';
+
+enum ModalType {
+  CANCEL_BOOKING = 'CANCEL_BOOKING',
+  CONTACT_HOST = 'CONTACT_HOST',
+}
+
+interface Props {
+  booking: Booking;
+  handleModalAction: (modal: ModalType) => void;
+}
+
+const TripCard = ({ booking, handleModalAction }: Props) => {
+  const { checkInDate, checkOutDate, id, listing, status } = booking;
+  const { addressLine1, addressLine2, city, country, lat, lng, postalCode, state } = listing;
+  const displayStatus = getUserBookingDisplayStatus(status);
+  return (
+    <Card className="mb-5 flex-fill" key={booking.id}>
+      <div className="embed-responsive embed-responsive-4by3">
+        <div className="embed-responsive-item">
+          <CardImg className="h-100" top src={booking.listing.listingPicUrl} alt="Listing Cover Photo" />
+        </div>
+      </div>
+      <CardBody>
+        <CardTitle className="h5 font-weight-normal mb-3">{booking.listing.title}</CardTitle>
+        <CardSubtitle className="small mb-3">
+          {addressLine1
+            ? formatAddress(addressLine1, addressLine2, city, state, country, postalCode)
+            : formatGeolocationAddress({ lat, lng, city, country })}
+        </CardSubtitle>
+        <Row>
+          <Col xs="6">
+            <small>Check-in:</small>
+            <h6>{formatSingleDate(checkInDate)}</h6>
+          </Col>
+          <Col xs="6">
+            <small>Check-out:</small>
+            <h6>{formatSingleDate(checkOutDate)}</h6>
+          </Col>
+        </Row>
+        <CardText>Status: {displayStatus}</CardText>
+        <CardText className="small">
+          Booking: <span>{id}</span>
+        </CardText>
+      </CardBody>
+      <CardFooter className="text-center">
+        <Row className="align-items-center">
+          <Col
+            xs="4"
+            onClick={() => handleModalAction(ModalType.CONTACT_HOST)}
```

Comment:
could this be `onClick={handleOpenContactHostModal}` and the method call `handleModalAction(ModalType.CONTACT_HOST)` ?

Reply:
could do. since there's also a `CANCEL_BOOKING` modal, that would require a `handleOpenCancelBookingModal` to be passed into this component as well. If it's conventionally better to use specific handle names, I'm all for it.

Reply:
i think using a method can ensure better enforcement vs a parameter.

Reply:
i see. so the component doesn't have to know about the enum. will change